### DISCUSSION TITLE
feat: Do you have any plans to implement borderless windows?

### DISCUSSION THREAD

**Discussion by xiusin** [ttytm/webview#10] [Q&A]
less windows can make windows look better. I vaguely remember that `wails` and `tarui` have similar functions.

**Comment by ttytm** [comment]
Hey @xiusin,

I enabled Discussion for this repo, I think this issue fits good here so I converted it.

---

The short answer is that it is not planned at the moment.

To elaborate:
The binding was created with the intention of keeping it low-level and closely aligned with its webview C API.
A library with a higher level API that is using this binding and has all kind of help functions to reduce boilerplate might evolve with time but for now the plan is:

- If standard functionalities like #6 cannot be handled properly without being integrated into the library, it is planned to add them to the library.
- If they can be handled without a library integration via builtin V, a simple example of them should be provided so users can handle them how they fit best per application.
- Not planned are platform-specific features that are not implemented upstream or in related low-level webview bindings in other languages.

  The reason for this is that it would open a barrel with a deep bottom in terms of new things that need to be supported, and it might require introducing breaking changes when alterations to related functionalities are made upstream.
  I'm concerned that this would result in a library with half-ass features that I will not be able to support. So for now, I prefer to keep it as simple as possible while integrating essential features.

  I probably should add something like this as a "Goals/Non-Goals" section in the readme or create a roadmap.

Regarding the borderless option, I'm fully with you, I also like it without borders. Depending on the platform, the compositor handles window titlebars and borders it's not always the application. I will have to look into it as things evolve to better tell how this can be best managed with webview.

**Comment by ttytm** [comment]
@xiusin to understand correctly, when you mean borderless do you mean hiding the titlebar? Or do you run into other borders that are created?

**Reply by xiusin** [reply]
@ttytm Yes, it's the meaning I want to express. Just like the game startup window。
